crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1175,32 +1174,38 @@ where
                             // to re-submit this transaction, we just returned the ready made
                             // certificate. A certificate is only valid if it's formed in the
                             // current epoch.
-                            Ok(VerifiedTransactionInfoResponse {
-                                certified_transaction: Some(inner_certificate),
-                                ..
-                            }) if inner_certificate.epoch() == self.committee.epoch  => {
-                                // A validator could return a certificate from an epoch that's
-                                // different from what the authority aggregator is expecting.
-                                // In that case, we should not accept that certificate.
-                                debug!(tx_digest = ?tx_digest, name=?name.concise(), weight, "Received prev certificate from validator handle_transaction");
-                                state.certificate = Some(inner_certificate);
-                            }
 
-                            // If we didn't match the above case but here, it means that we have
-                            // a cert from a different epoch, and also have effects (i.e. already
-                            // executed), we can accept the certificate if we get 2f+1 effects.
-                            // It's an proof that the transaction has already been finalized
-                            // in a different epoch, and hence it's ok to reuse the old certificate.
+                            // Note that validator signs a certificate happened in a past epoch
+                            // with its current epoch id, so the case where we have a valid TxCert
+                            // but a mismatched epoch id is very very rare:
+                            // 1. If the cert's epoch is bigger, SafeClient will scream and we 
```

Comment:
SafeClient won't scream as long as it is able to find the committee in the committee store.

Reply:
that's what I mean by scream, returning `SuiError::MissingCommitteeAtEpoch`, will make it clear